### PR TITLE
fix(stepfunctions): allow disable x-ray

### PR DESCRIPTION
**Pull request by watany-dev**
### Issue # (if applicable)

Closes [#30796 .](https://github.com/aws/aws-cdk/issues/30796)

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  comment: 'a super cool state machine',
+  tracingEnabled: false,
+});
+
+// stateMachine.grantRead(role);
+// stateMachine.grant(role, 'states:SendTaskSuccess');
+
+app.synth();
```
Comment (ChakshuGupta13) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts:25]: a snapshot is needed for this integration test file - which is why some checks are failing. this snapshot will confirm that these changes are deployable to an AWS account. is it possible to add that snapshot?

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+//   assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+// });
+
+const stateMachine = new sfn.StateMachine(stack, 'StateMachine', {
+  comment: 'a super cool state machine',
+  tracingEnabled: false,
+});
+
+// stateMachine.grantRead(role);
+// stateMachine.grant(role, 'states:SendTaskSuccess');
```
Comment (ChakshuGupta13) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts:23]: doubt: why are these code comments needed?

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * -- aws stepfunctions describe-state-machine --state-machine-arn <stack-output> has a status of `ACTIVE`
+ * -- aws iam get-role-policy --role-name <role-name> --policy-name <policy-name> has all actions mapped to respective resources.
+ */
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-stepfunctions-integ');
+
+// const role = new iam.Role(stack, 'Role', {
+//   assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+// });
```
Comment (ChakshuGupta13) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts:15]: doubt: why are these code comments needed?

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+/*
+ * Stack verification steps:
+ *
+ * -- aws stepfunctions describe-state-machine --state-machine-arn <stack-output> has a status of `ACTIVE`
+ * -- aws iam get-role-policy --role-name <role-name> --policy-name <policy-name> has all actions mapped to respective resources.
+ */
```
Comment (ChakshuGupta13) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts:9]: doubt: how will these steps help developers?

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
```
Comment (ChakshuGupta13) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions/test/integ.state-machine-disable-xray.ts:3]: nit: an empty line after import may be better?

**Review by ChakshuGupta13 — CHANGES_REQUESTED** [review]

**Comment by watany-dev** [comment]
@ChakshuGupta13 
Sorry for the wait, the absence of snapshot has been resolved.

**Review by ChakshuGupta13 — APPROVED** [review]

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Review by TheRealAmazonKendra — APPROVED** [review]
Thanks for the fix!

**Pull request merged**